packages/next/next-server/server/lib/storage.ts

```diff
@@ -0,0 +1,19 @@
+import { promises } from 'fs'
+
+export interface StorageProfiderInterface {
+  readValue: (key: string) => Promise<string>
+  write: (key: string, value: string | undefined) => Promise<void>
+}
+
+export class FileSystemStorageProvider implements StorageProfiderInterface {
+  async readValue(key: string): Promise<string> {
+    return await promises.readFile(key, 'utf8')
+  }
+
+  async write(key: string, value: string | undefined) {
+    await promises.mkdir(key.substring(0, key.lastIndexOf('/')), {
```

Comment:
Testing this pull request and apparently this makes the directory to be an empty string and causing the `write()` to fail each time it tries to save the files to disk.

```
Failed to update prerender files for /test [Error: ENOENT: no such file or directory, mkdir ''] {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'mkdir',
  path: ''
}
```
The file should require `path` and the line should be `path.dirname(key)` to achieve the same result as the current implementation.

Also, `mkdir` could be put in a different method and called before trying to write the HTML and JSON, as it will be called twice for each page.